app/controllers/api/volume_mappings_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Api
+  class VolumeMappingsController < BaseController
+    def refresh_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" if id.blank?
+
+      ensure_resource_exists(type, id) if single_resource?
+
+      api_action(type, id) do |klass|
+        volume_mapping = resource_search(id, type, klass)
+        api_log_info("Refreshing #{volume_mapping_ident(volume_mapping)}")
+        refresh_volume_mapping(volume_mapping)
+      end
+    end
+
+    def create_resource(_type, _id = nil, data = {})
+      ext_management_system = ExtManagementSystem.find(data['ems_id'])
+      raise BadRequestError "Volume Mappings not supported for #{ext_management_system.class}" unless ext_management_system.class::VolumeMapping.supports_create?
```

Comment:
`ext_management_system.class::VolumeMapping.supports_create?`

```
klass = VolumeMapping.class_by_ems(ems)
raise BadRequestError, klass.unsupported_reason(:create) unless klass.supports?(:create)
```
Prefer to use `VolumeMapping.class_for_ems(ems).supports_create?`

Reply:
Will do. Why don't we also check that class_by_ems didn't return a nil?

Reply:
`class_by_ems` will only return `nil` if the `ems` passed in is nil and we're doing a `.find(id)` which will raise an exception if it isn't found.

Reply:
Right, or if the ems doesn't have a VolumeMapping sub-type. 
So wouldn't we want to raise an exception with a meaningful method, such as `EMS of type XXX don't support creation of YYY`?